EventFilter/GEMRawToDigi/plugins/GEMRawToDigiModule.cc

```diff
@@ -0,0 +1,195 @@
+/** \unpacker for gem
+ *  \author J. Lee - UoS
+ */
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+
+#include "EventFilter/GEMRawToDigi/plugins/GEMRawToDigiModule.h"
+
+using namespace gem;
+
+GEMRawToDigiModule::GEMRawToDigiModule(const edm::ParameterSet & pset)
+{
+  fed_token = consumes<FEDRawDataCollection>( pset.getParameter<edm::InputTag>("InputLabel") );  
+  useDBEMap_ = pset.getParameter<bool>("useDBEMap");
+  produces<GEMDigiCollection>(); 
+  unpackStatusDigis_ = pset.getParameter<bool>("UnpackStatusDigis");
+  if (unpackStatusDigis_){
+    produces<GEMVfatStatusDigiCollection>("vfatStatus");
+    produces<GEMGEBStatusDigiCollection>("GEBStatus");
+  }
+}
+
+void GEMRawToDigiModule::fillDescriptions(edm::ConfigurationDescriptions & descriptions)
+{
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("InputLabel", edm::InputTag("rawDataCollector")); 
+  desc.add<bool>("useDBEMap", false); 
+}
+
+void GEMRawToDigiModule::beginRun(edm::Run const&, edm::EventSetup const& iSetup)
+{
+  if (useDBEMap_){
+    edm::ESHandle<GEMEMap> gemEMap;
+    iSetup.get<GEMEMapRcd>().get(gemEMap);
+    m_gemEMap = gemEMap.product();
+    m_gemROMap = m_gemEMap->convert();
+  }
+  else {
+    // no eMap, using dummy
+    m_gemEMap = new GEMEMap();
```

Comment:
where is m_gemEMap deleted

Reply:
It is deleted in the endRun method. But it would be better to use a smart pointer for this.